app/org/sagebionetworks/bridge/services/ScheduledActivityService.java

```diff
@@ -214,6 +229,29 @@ public void deleteActivitiesForUser(String healthCode) {
             .collect(toImmutableList());
     }
     
+    /**
+     * If the client data is being added or removed, or if it is different, then the activity is being 
+     * updated.
+     */
+    private boolean hasUpdatedClientData(ScheduledActivity schActivity, ScheduledActivity dbActivity) {
```

Comment:
Is it worth adding unit test cases for

- both null
- added
- removed
- changed
- same